segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/Ledgers.java

```diff
@@ -88,7 +88,7 @@ static LedgerHandle openFence(long ledgerId, BookKeeper bookKeeper, BookKeeperCo
     static LedgerHandle openRead(long ledgerId, BookKeeper bookKeeper, BookKeeperConfig config) throws DurableDataLogException {
         try {
             return Exceptions.handleInterruptedCall(
-                    () -> bookKeeper.openLedgerNoRecovery(ledgerId, LEDGER_DIGEST_TYPE, config.getBKPassword()));
+                    () -> bookKeeper.openLedgerNoRecovery(ledgerId, config.getDigestType(), config.getBKPassword()));
```

Comment:
I suggest to enable digestTypeAutodetection so you don't  need to care about it on the reader side and your code will be compatible with rolling upgrades